packages/react-core/src/components/Chip/__tests__/Chip.test.tsx

```diff
@@ -1,59 +1,234 @@
   });
 
-  test('closable', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" id="chip_one">
-        Chip
+  test(`Renders a badge when badge prop is passed to ${chipType}`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} badge={<div>Badge content</div>}>
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByText('Badge content')).toBeVisible();
   });
 
-  test('closable with tooltip', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" id="chip_one">
-        1234567890123456789
+  test(`Renders with div container on ${chipType} by default`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container').tagName).toBe('DIV');
   });
 
-  test('readonly', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" isReadOnly>
-        4 more
+  test(`Renders with custom container on ${chipType} when component prop is passed`, () => {
+    render(
+      <Chip component="button" isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container').tagName).toBe('BUTTON');
   });
 
-  test('custom max-width text', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" textMaxWidth="100px">
-        4 more
+  test(`Renders maxWidth css var in style attribute when textMaxWidth is passed for ${chipType}`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container" textMaxWidth="10px">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container')).toHaveAttribute('style', '--pf-v5-c-chip__text--MaxWidth: 10px;');
   });
 
-  test("with role='gridcell'", () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" role="gridcell">
-        I'm a roled chip
+  test(`Spreads additional props to container for ${chipType}`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container" role="button">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container')).toHaveAttribute('role', 'button');
   });
 });
+
+test(`Renders id prop on ${styles.chipText} container for default chip`, () => {
+  render(<Chip id="custom-id">Chip text</Chip>);
+
+  expect(screen.getByText('Chip text')).toHaveAttribute('id', 'custom-id');
+});
+
+test(`Does not render id prop on ${styles.chipText} container for overflow chip`, () => {
+  render(
+    <Chip isOverflowChip id="custom-id">
+      Chip text
+    </Chip>
+  );
+
+  expect(screen.getByText('Chip text')).not.toHaveAttribute('id');
+});
```

Comment:
Yeah right now the `id` is placed on the `chip__text` element normally, I assume (at least partially?) because the chip action's close button has its aria-labelledby set to the chip text and the close button itself. So at least here we do need the `id` to be applied to the `chip__text`. Whether it should be placed there for an overflow chip as well... We don't necessarily *need* it there, so having it on the top level isn't an issue per se. Though it is odd for it to be rendered in different elements.

If we wanted things to be more consistent, we could think about applying the base `id` (either whatever is passed in or the internally generated random one) to the top level element, then for the `chip__text` element taking the base id + `chip-text` or something.

Reply:
@tlabaj WDYT?

Reply:
So it looks like right now the id would be at the top element and the chip__text element when it is overflow..  wouldn't that be a duplicate id? I am surprised a11y test are not complaining?

Reply:
A follow up or fixing here is fine.  I would not consider the PR blocked by not resolving that preexisting issue.

Reply:
Opened https://github.com/patternfly/patternfly-react/issues/9764 as a followup just to keep it separate from test updates